webui/react/src/components/kit/Icon.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { CSSProperties } from 'react';
+
+import Tooltip from 'components/kit/Tooltip';
+
+import css from './Icon.module.scss';
+
+export const IconSizeArray = [
+  'tiny',
+  'small',
+  'medium',
+  'large',
+  'big',
+  'great',
+  'huge',
+  'enormous',
+  'giant',
+  'jumbo',
+  'mega',
+] as const;
+
+export type IconSize = (typeof IconSizeArray)[number];
+
+export const IconNameArray = [
+  'home',
+  'dai-logo',
+  'arrow-left',
+  'arrow-right',
+  'add-small',
+  'close-small',
+  'search',
+  'arrow-down',
+  'arrow-up',
+  'cancelled',
+  'group',
+  'warning-large',
+  'steering-wheel',
+  'workspaces',
+  'archive',
+  'queue',
+  'model',
+  'fork',
+  'pause',
+  'play',
+  'stop',
+  'reset',
+  'undo',
+  'learning',
+  'heat',
+  'scatter-plot',
+  'parcoords',
+  'pencil',
+  'settings',
+  'filter',
+  'docs',
+  'power',
+  'close',
+  'dashboard',
+  'checkmark',
+  'cloud',
+  'document',
+  'logs',
+  'tasks',
+  'checkpoint',
+  'download',
+  'debug',
+  'error',
+  'warning',
+  'info',
+  'clipboard',
+  'fullscreen',
+  'eye-close',
+  'eye-open',
+  'user',
+  'jupyter-lab',
+  'lock',
+  'user-small',
+  'popout',
+  'spinner',
+  'collapse',
+  'expand',
+  'tensorboard',
+  'cluster',
+  'command',
+  'experiment',
+  'grid',
+  'list',
+  'notebook',
+  'overflow-horizontal',
+  'overflow-vertical',
+  'shell',
+  'star',
+  'tensor-board',
+  'searcher-random',
+  'searcher-grid',
+  'searcher-adaptive',
+  'critical',
+  'trace',
+  'webhooks',
+] as const;
+
+export type IconName = (typeof IconNameArray)[number];
+
+export interface Props {
+  name?: IconName;
+  size?: IconSize;
+  style?: CSSProperties;
```

Comment:
We shouldn't allow arbitrary styles.

Reply:
suggestion: a `state` prop for `'cancel'|'success'|'error'` that applies the appropriate css var instead of allowing any color? 

maybe also just a `heavy` boolean prop or something if necessary

could add such props to testing/variations displayed on UI kit page.

Reply:
No, the three icons that have `fontWeight: 900` set are checkmark, error, and cancelled. That file seems to be the only place we use the cancelled icon, but the other two are used elsewhere without font weight being set. I can add a color prop to Icon, though they're all in the form of SCSS variables like `var(--theme-status-error)`, is that alright?

Reply:
Ok, I took a look at the bolded usages, we can unbold those. They actually render a little weird when bolded anyway. If we want them thicker in the future we should make them thicker consistently.

For color taking color as a color _name_ is totally fine. That can either be something like John suggests or you can take `'theme-status-error' | 'theme-status-success' | ...`. Exposing available colors as an enum in the UI Kit is something we probably want to do anyway.